main.py

```diff
@@ -271,10 +282,13 @@ def quantize_aq(
 
                 weight_avg_bits = quantized_weight.estimate_nbits_per_parameter()
                 overall_bits += int(weight_avg_bits * torch.numel(aq_handlers[sublayer_name].layer.weight.data))
-                model_number_of_params += torch.numel(aq_handlers[sublayer_name].layer.weight.data)
-                print("curent_avg_bits", overall_bits / model_number_of_params)
+                number_of_quantized_params += torch.numel(aq_handlers[sublayer_name].layer.weight.data)
+                print("curent_avg_bits", overall_bits / number_of_quantized_params)
                 quantizers["model.layers.%d.%s" % (layer_index, sublayer_name)] = ()  # to be updated
 
+            del aq_handlers
```

Comment:
Rationale: this saves a bit of memory by deleting XTX after we no longer need them. We can use this to fit ever so slightly larger models.